### PR TITLE
Feature/core/copy crypto files

### PR DESCRIPTION
**Pull request by TalhaAli00**

### CODE REVIEW

**Comment by jakobuid** [comment]
@TalhaAli00 
Change a branch that you want to merge into. You have over 200 changes :P

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/Messages.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.walletconnect.android_core.common.exceptions
+package com.walletconnect.android_core.common
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/common/Messages.kt:1]: should we add JvmSynthetic annotation here?

File: android_core/src/main/kotlin/com/walletconnect/android_core/crypto/data/repository/JwtRepositoryAndroid.kt [hunk 2]
```diff
@@ -45,6 +46,10 @@ internal class JwtRepository(private val keyChain: KeyStore) {
         return encodeJWT(JWT_IRN_HEADER, payload, signature)
     }
 
+    override fun setKeyPair(key: String, privateKey: PrivateKey, publicKey: PublicKey) {
+        TODO("Not yet implemented")
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/crypto/data/repository/JwtRepositoryAndroid.kt:50]: We need to save keys here to keyChain. 
Previous implementation:

`        keyChain.setKeys(KEY_DID_KEYPAIR, privateKey, publicKey)
`

File: android_core/src/main/kotlin/com/walletconnect/android_core/crypto/KeyManagementRepository.kt [hunk 3]
```diff
@@ -1,12 +1,10 @@
...
-
-package com.walletconnect.sign.crypto
+package com.walletconnect.android_core.crypto
 
 import com.walletconnect.foundation.common.model.PublicKey
 import com.walletconnect.foundation.common.model.SymmetricKey
 import com.walletconnect.foundation.common.model.Topic
 
-internal interface KeyManagementRepository {
+interface KeyManagementRepository {
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/crypto/KeyManagementRepository.kt:7]: Can we remove comment at the bottom on the class

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt [hunk 4]
```diff
@@ -14,35 +15,42 @@ import com.walletconnect.android_core.network.data.connection.controller.Connect
 import com.walletconnect.android_core.network.data.connection.lifecycle.ManualConnectionLifecycle
 import com.walletconnect.android_core.network.domain.RelayClient
 import com.walletconnect.foundation.network.data.adapter.FlowStreamAdapter
+import okhttp3.Interceptor
 import com.walletconnect.foundation.di.networkModule as foundationNetworkModule
 import okhttp3.OkHttpClient
+import org.bouncycastle.asn1.x500.style.RFC4519Style.name
```
Comment (Elyniss) [inline android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:21]: Please run optimise imports?

File: foundation/src/main/kotlin/com/walletconnect/foundation/di/CryptoModule.kt [hunk 5]
```diff
@@ -6,7 +6,7 @@ import com.walletconnect.foundation.crypto.data.repository.BaseJwtRepository
 import com.walletconnect.foundation.crypto.data.repository.JwtRepository
 import org.koin.dsl.module
 
-fun cryptoModule() = module {
+internal fun cryptoModule() = module {
```
Comment (Elyniss) [inline foundation/src/main/kotlin/com/walletconnect/foundation/di/CryptoModule.kt:9]: please add JvmSyntethic annotation

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/common/model/vo/clientsync/session/payload/SessionEventVO.kt [hunk 6]
```diff
@@ -1,3 +1,5 @@
+@file:JvmSynthetic
```
Comment (Elyniss) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/common/model/vo/clientsync/session/payload/SessionEventVO.kt:1]: Should SessionEventVO be internal?

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt [hunk 7]
```diff
@@ -1,27 +1,30 @@
...
-import com.walletconnect.sign.common.adapters.RelayDOJsonRpcResultJsonAdapter
+import com.walletconnect.sign.common.adapters.JsonRpcResultAdapter
 import com.walletconnect.sign.common.adapters.SessionRequestVOJsonAdapter
 import com.walletconnect.sign.common.model.vo.clientsync.session.payload.SessionRequestVO
 import org.koin.core.qualifier.named
 import org.koin.dsl.module
 import kotlin.reflect.jvm.jvmName
 import com.walletconnect.android_core.di.commonModule as androidCoreCommonModule
 
+@JvmSynthetic
```
Comment (Elyniss) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt:16]: Shouldn't common module be internal?

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt (hunk 4 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:21]: done

**Review by Elyniss — COMMENTED** [review]
Nice catch with code duplication

File: foundation/src/main/kotlin/com/walletconnect/foundation/di/CryptoModule.kt (hunk 5 above)
Comment (TalhaAli00) [reply foundation/src/main/kotlin/com/walletconnect/foundation/di/CryptoModule.kt:9]: done

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt (hunk 7 above)
Comment (TalhaAli00) [reply signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt:16]: yes, it should

File: android_core/src/main/kotlin/com/walletconnect/android_core/crypto/data/repository/JwtRepositoryAndroid.kt (hunk 2 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/crypto/data/repository/JwtRepositoryAndroid.kt]: fixed

File: android_core/src/main/kotlin/com/walletconnect/android_core/common/Messages.kt (hunk 1 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/common/Messages.kt:1]: added annotation

File: android_core/src/main/kotlin/com/walletconnect/android_core/crypto/KeyManagementRepository.kt (hunk 3 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/crypto/KeyManagementRepository.kt:7]: comment removed

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt [hunk 8]
```diff
@@ -60,17 +68,17 @@ fun networkModule(serverUrl: String, jwt: String, connectionType: ConnectionType
         }
     }
 
+    single { ConnectivityState(androidApplication()) }
+
     single {
         Scarlet.Builder()
             .backoffStrategy(get<LinearBackoffStrategy>())
-            .webSocketFactory(get<OkHttpClient>().newWebSocketFactory("$serverUrl&auth=$jwt"))
+            .webSocketFactory(get<OkHttpClient>(named("android-core")).newWebSocketFactory("$serverUrl&auth=$jwt"))
```
Comment (jakobuid) [inline android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:76]: Can we extract "android-core" to DI tags?

File: android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/domain/JsonRpcInteractorAbstract.kt [hunk 9]
```diff
@@ -1,28 +1,26 @@
...
+import com.walletconnect.android_core.network.data.connection.ConnectivityState
 import com.walletconnect.android_core.storage.JsonRpcHistory
 import com.walletconnect.android_core.utils.Logger
-import com.walletconnect.android_core.common.exceptions.client.WalletConnectException
 import com.walletconnect.foundation.common.model.SubscriptionId
 import com.walletconnect.foundation.common.model.Topic
 import com.walletconnect.foundation.network.model.Relay
-import com.walletconnect.sign.crypto.Codec
 import com.walletconnect.utils.Empty
 import kotlinx.coroutines.CoroutineExceptionHandler
```
Comment (jakobuid) [inline android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/domain/JsonRpcInteractorAbstract.kt:25]: Let's remove todo coment on line 35

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 10]
```diff
@@ -114,6 +114,7 @@ fun DependencyHandlerScope.jUnit5() {
     "testImplementation"("org.junit.jupiter:junit-jupiter-api:$jUnit5Version")
     "testRuntimeOnly"("org.junit.jupiter:junit-jupiter-engine:$jUnit5Version")
     "testImplementation"("org.jetbrains.kotlin:kotlin-test-junit5:$kotlinVersion")
+
```
Comment (jakobuid) [inline buildSrc/src/main/kotlin/Dependencies.kt:117]: Unnecessary empty line

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt [hunk 11]
```diff
@@ -1,27 +1,30 @@
...
 fun commonModule() = module {
 
     includes(androidCoreCommonModule())
 
     single {
         get<Moshi>(named("android_core"))
```
Comment (jakobuid) [inline signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt:22]: I saw in other modules "android-core" tag. Let's use one way pls

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt (hunk 8 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:76]: are you asking for just android-core? Or to use the DITags pattern everywhere?

**Review by jakobuid — APPROVED** [review]

File: android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/domain/JsonRpcInteractorAbstract.kt (hunk 9 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/json_rpc/domain/JsonRpcInteractorAbstract.kt:25]: done

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt (hunk 8 above)
Comment (jakobuid) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:76]: I was thinking about DITags pattern

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 10 above)
Comment (TalhaAli00) [reply buildSrc/src/main/kotlin/Dependencies.kt:117]: removed

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt (hunk 11 above)
Comment (TalhaAli00) [reply signSDK/sign/src/main/kotlin/com/walletconnect/sign/di/CommonModule.kt:22]: good catch

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt (hunk 8 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:76]: we can but each module would need it's own DITags enum class. Kotlin doesn't allow open enum classes so we can't build on it at each layer

Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt:76]: I guess having different DITags classes isn't too bad. It'll keep us from making mistakes with Strings

**Review by Elyniss — APPROVED** [review]

File: android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt (hunk 8 above)
Comment (TalhaAli00) [reply android_core/src/main/kotlin/com/walletconnect/android_core/di/NetworkModule.kt]: added enum tags

**Pull request merged**